src/data/disks.rs

```diff
@@ -145,8 +163,47 @@ extern "C" {
 
 #[cfg(not(windows))]
 fn load_disks(canonicalize_paths: bool) -> Vec<Disk> {
-    sysinfo::Disks::new_with_refreshed_list()
-        .iter()
-        .map(|d| Disk::from_sysinfo_disk(d, canonicalize_paths))
-        .collect()
+
+    let mut result: Vec<Disk> = Vec::new();
+    let mut seen_mount_points = std::collections::HashSet::new();
+
+    // Use sysinfo to get system disks
+    let disks = sysinfo::Disks::new_with_refreshed_list();
+    for disk in &disks {
+        let mount_point = disk.mount_point();
+        if mount_point == Path::new("/") {
+            // Skip the root system drive
+            continue;
+        }
+
+        // Ensure no duplicates by checking the mount point
+        if !seen_mount_points.insert(mount_point.to_path_buf()) {
+            continue;
+        }
```

Comment:
What is the advantage of using the `seen_mount_points` instead of just checking the entries in `result`?

Reply:
I simplified the code a bit, is that what you mean?

Reply:
You should be able to completely remove the `seen_mount_points` by replacing:
```rust
        // Ensure no duplicates by checking the mount point
        if !seen_mount_points.insert(mount_point.to_path_buf()) {
            continue;
        }
```
with:
```rust
    // Ensure no duplicates by checking the mount point
    if result.iter().any(|d| d.mount_point == mount_point) {
        continue;
    }
```

Reply:
no, this also leaves out `Macintosh HD`

<img width="765" alt="Bildschirmfoto 2024-11-18 um 22 23 08" src="https://github.com/user-attachments/assets/b5b638b6-bc54-40d3-b1e9-26820fe4565d">

Reply:
thats weird. then let's leave it like that for now